mods/farming/nodes.lua

```diff
@@ -175,68 +191,75 @@ minetest.register_abm({
 	interval = 15,
 	chance = 4,
 	action = function(pos, node)
-		local n_def = minetest.registered_nodes[node.name] or nil
-		local wet = n_def.soil.wet or nil
-		local base = n_def.soil.base or nil
-		local dry = n_def.soil.dry or nil
-		if not n_def or not n_def.soil or not wet or not base or not dry then
-			return
-		end
-
 		pos.y = pos.y + 1
 		local nn = minetest.get_node_or_nil(pos)
-		if not nn or not nn.name then
+		if not nn then
 			return
 		end
-		local nn_def = minetest.registered_nodes[nn.name] or nil
+		local nn_def = minetest.registered_nodes[nn.name]
 		pos.y = pos.y - 1
 
-		if nn_def and nn_def.walkable and minetest.get_item_group(nn.name, "plant") == 0 then
-			minetest.set_node(pos, {name = base})
+		local soil = minetest.registered_nodes[node.name].soil
+		assert(soil and soil.wet and soil.base and soil.dry,
+			"Field " .. node.name .. " does not have all soil properties.")
+
+		if nn_def and nn_def.walkable and
+				minetest.get_item_group(nn.name, "plant") == 0 then
+			node.name = soil.base
+			minetest.set_node(pos, node)
 			return
 		end
-		-- check if there is water nearby
+
 		local wet_lvl = minetest.get_item_group(node.name, "wet")
+		-- Make the node wet if water is near it
 		if minetest.find_node_near(pos, 3, {"group:water"}) then
-			-- if it is dry soil and not base node, turn it into wet soil
+			-- If it is dry soil and not base node, turn it into wet soil
 			if wet_lvl == 0 then
-				minetest.set_node(pos, {name = wet})
+				node.name = soil.wet
+				minetest.set_node(pos, node)
 			end
-		else
-			-- only turn back if there are no unloaded blocks (and therefore
-			-- possible water sources) nearby
-			if not minetest.find_node_near(pos, 3, {"ignore"}) then
-				-- turn it back into base if it is already dry
-				if wet_lvl == 0 then
-					-- only turn it back if there is no plant/seed on top of it
-					if minetest.get_item_group(nn.name, "plant") == 0 and minetest.get_item_group(nn.name, "seed") == 0 then
-						minetest.set_node(pos, {name = base})
-					end
-
-				-- if its wet turn it back into dry soil
-				elseif wet_lvl == 1 then
-					minetest.set_node(pos, {name = dry})
-				end
+			return
+		end
+
+		-- Only dry out if there are no unloaded blocks (and therefore
+		-- possible water sources) nearby
+		if minetest.find_node_near(pos, 3, {"ignore"}) then
+			return
+		end
+
+		-- Turn it back into base if it is already dry and no plant/seed
+		-- is on top of it
+		if wet_lvl == 0 then
+			if minetest.get_item_group(nn.name, "plant") == 0 and
+					minetest.get_item_group(nn.name, "seed") == 0 then
+				node.name = soil.base
+				minetest.set_node(pos, node)
 			end
+
+		-- If it is wet turn it back into dry soil
+		elseif wet_lvl == 1 then
+			node.name = soil.dry
+			minetest.set_node(pos, node)
 		end
 	end,
 })
 
 
+local defchange = {drop = {
```

Comment:
If you're already changing this, please move `drop` to the next line.

Reply:
done